src/libopensc/card-entersafe.c

```diff
@@ -345,14 +378,11 @@ static int entersafe_mac_apdu(sc_card_t *card, sc_apdu_t *apdu,
 	 apdu->datalen=apdu->lc;
 
 out:
-	 if(tmp)
-		  free(tmp);
-	 if(tmp_rounded)
-		  free(tmp_rounded);
-	 if (ctx) {
-		sc_evp_cipher_free(alg);
-		EVP_CIPHER_CTX_free(ctx);
-	}
+	sc_log_openssl(card->ctx);
```

Comment:
should this be called only if the `r != SC_OK`? Or is the code now quite when there is no error to report?

Reply:
Yes, thanks for noticing - after the change to `ERR_print_errors` it would print OpenSSL logging error when no errors are in the stack.